### DISCUSSION TITLE
Disabling image load animation for SwiftUI

### DISCUSSION THREAD

**Discussion by kaspesi** [onevcat/Kingfisher#2292] [General]
Trying to disable the image load animation when a image is loaded from the network and not the cache.

Ive tried setting `.fade(duration: 0.0)` but have only been able to remove this animation with the depreciated `.animation(nil)` modifier.

```
  var body: some View {
    KFImage.url(url, cacheKey: key)
//      .fade(duration: 0.0)
      .setProcessor(DownsamplingImageProcessor(size: CGSize(width: width ?? UIScreen.main.bounds.size.width, height:  height ?? UIScreen.main.bounds.size.height)))
      .placeholder {
        ProgressView()
      }

      .onSuccess { result in
        print("Image loaded successfully: \(result.cacheType)")
        print("Image Width: \(width)")
      }
  .animation(nil)
  }
}
```

This works for me, but was curious if theres a better way to do this.  Any help appricated!

**Comment by gg6126811-oss** [comment]
That’s an interesting issue. It sounds like the .fade(duration:) modifier isn’t disabling the animation when the image loads from the network, which is definitely frustrating. The .animation(nil) trick used to work well, but since it’s deprecated, there might not be a direct replacement yet. You could try applying .withAnimation(nil) in the onSuccess block or check if Kingfisher’s newer API supports a conditional fade for network vs. cache loads. Hopefully, they add a cleaner way to handle this in future updates.

**Comment by DonaldJSmith12** [comment]
You can disable the load animation by using Kingfisher’s transition options instead of the deprecated .animation(nil). Applying .fade(.none) on KFImage removes the loading animation for both cached and network images. Example: KFImage.url(url).fade(.none). This avoids warnings and gives consistent behavior.

**Comment by jerrycpoplar6556-max** [comment]
Kingfisher’s own fade effect is separate from SwiftUI’s animation system, so removing it there usually does the trick. If you’re still seeing a transition, make sure no parent view has implicit animations applied

**Comment by sultanahmed21** [comment]
I think tools and apps like this can be useful when connection or routing is the real issue, not the game itself. I came across [VIP Proxy Server Free Fire APK](https://proxyserverfreefire.com/) while looking into network-related fixes, and it explains things in a pretty straightforward way. At the end of the day, it really depends on the user’s setup and network.
